src/main/groovy/com/twcable/grabbit/jcr/JcrNodeDecorator.groovy

```diff
@@ -136,6 +136,41 @@ class JcrNodeDecorator {
     }
 
     /**
+     * Finds out nearest versionable ancestor for a node and performs a checkout
+    */
+    void checkoutNode() {
+        try {
+            JcrNodeDecorator decoratedVersionableAncestor = findVersionableAncestor();
+            if (decoratedVersionableAncestor && !decoratedVersionableAncestor.isCheckedOut()) {
+                decoratedVersionableAncestor.session.workspace.versionManager.checkout(decoratedVersionableAncestor.path);
+            }
+        }
+        catch (Exception exception) {
+            log.warn "Could not checkout node ${this.path}, ${exception.message}"
+        }
+    }
+
+    private JcrNodeDecorator findVersionableAncestor() throws RepositoryException {
+        if (isVersionable()) {
+            return this;
+        }
+        try {
+            JcrNodeDecorator parentDecoratedNode = new JcrNodeDecorator(this.parent);
+            return parentDecoratedNode.findVersionableAncestor();
+        } catch (ItemNotFoundException e) {
+            return null;
+        }
+    }
+
+    /**
+     * mix:simpleVersionable cannot be checked in. Throws javax.jcr.UnsupportedRepositoryOperationException
```

Comment:
I'd say add both. We idol worship the JSR-283 before Jackrabbit 🙏 

If Jackrabbit decides to be more compliant, then or application won't break.